bundles/CoreBundle/Command/GenerateStaticPagesCommand.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Bundle\CoreBundle\Command;
+
+use Pimcore\Console\AbstractCommand;
+use Pimcore\Document\StaticPageGenerator;
+use Pimcore\Model\DataObject;
+use Pimcore\Model\DataObject\ClassDefinition;
+use Pimcore\Model\DataObject\ClassDefinition\ClassDefinitionManager;
+use Pimcore\Model\Document;
+use Symfony\Component\Console\Helper\ProgressBar;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * @internal
+ */
+class GenerateStaticPagesCommand extends AbstractCommand
+{
+    protected StaticPageGenerator $staticPageGenerator;
+
+    public function __construct(StaticPageGenerator $staticPageGenerator)
+    {
+        parent::__construct();
+
+        $this->staticPageGenerator = $staticPageGenerator;
+    }
+
+    protected function configure()
+    {
+        $this
+            ->setName('pimcore:generate-static-pages')
+            ->setDescription('Regenerate static Pages')
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $listing = new Document\Listing();
+        $listing->setCondition("type = 'page'");
+        $listing->setOrderKey('id');
+        $listing->setOrder('DESC');
```

Comment:
@dpfaffenbauer please add option to filter documents by parent path. As mentioned in issue https://github.com/pimcore/pimcore/issues/8196 "_command that generates the static pages incl. the possibility to filter pages e.g. by parent path_"

Reply:
done